tests/components/application-connector/README.md

```diff
@@ -2,65 +2,172 @@
+
+The tests are executed as a Kubernetes Job on a Kyma cluster, 
+where the tested Application Gateway is installed. 
+The test job and mock application deployment are in the `test` Namespace. 
 
 ![Application Gateway tests architecture](./assets/app-gateway-tests-architecture.svg)
 
-The test cases are defined as services in the Application CR.
+## Building
 
-The whole test setup is deployed into the Kyma cluster with the Helm template executed in the Makefile by the command `test-gateway`.
+To build **and push** Docker images of tests and mock application:
 
-Image versions and the external service name used during testing can be set up in the Helm chart values file `k8s/gateway-test/values.yaml`.
+``` sh
+./scripts/local-build.sh {DOCKER_TAG} {DOCKER_PUSH_REPOSITORY}
+```
+This will build the following images:
+- `{DOCKER_PUSH_REPOSITORY}/gateway-test:{DOCKER_TAG}`
+- `{DOCKER_PUSH_REPOSITORY}/mock-app:{DOCKER_TAG}`
 
-### Local build of test images
+## Running
 
-1. In the main folder build the test images.
-   ```bash
-   ./scripts/local-build.sh {DOCKER_TAG} {DOCKER_PUSH_REPOSITORY}
-   ```
+Tests can be run on any Kyma cluster with Application Gateway.
 
-### Local execution
+### Deploy Kyma cluster locally
 
 1. Provision a local Kubernetes cluster with k3d:
-
-   ```shell
+   ```sh
    kyma provision k3d
    ```
 
-2. Install the minimal set of components required to run Application Gateway for either Kyma OS or SKR:
+1. Install the minimal set of components required to run Application Gateway for either Kyma OS or SKR:
+
+    <div tabs name="Kyma flavor" group="minimal-kyma-installation">
+    <details open>
+    <summary label="OS">
+    Kyma OS
+    </summary>
+
+    ```sh
+    kyma deploy --components-file ./resources/installation-config/mini-kyma-os.yaml
+    ```
+
+    </details>
+    <details>
+    <summary label="SKR">
+    SKR
+    </summary>
+
+    ```bash
+    kyma deploy --components-file ./resources/installation-config/mini-kyma-skr.yaml 
+    ```
+
+    </details>
+    </div>
+
+    >**TIP:** More on Kyma installation can be found in [official Kyma documentation](https://kyma-project.io/docs/kyma/latest/02-get-started/01-quick-install/#install-kyma)
+
+### Run the tests
+
+``` sh
+make test-gateway
+```
+
+By default tests clean up after themselves, removing all created resources
+and `test` Namespace.
+
+> **CAUTION:** This might override or remove existing resources, 
+> if names of already existing resources collide with names used by tests
+
+## Debugging
 
-   <div tabs name="Kyma flavor" group="minimal-kyma-installation">
-      <details open>
-      <summary label="OS">
-      Kyma OS
-      </summary>
+### Running locally
 
-   ```bash
-   kyma deploy --components-file ./resources/installation-config/mini-kyma-os.yaml
+The Test Job must run on a cluster, because of the way it accesses Application CRs.
```

Comment:
I do not explain how to run it locally and it'll just crash instantly, so it should be hard to break anything by doing it incorrectly, I don't know if it's caution-worthy. It just won't start

Reply:
I was thinking more along the lines of this being important enough to draw special attention to it -- it might not do any harm, but the user might not understand why it crashes when they run it locally and think that there's a problem with the tests, and not with the way they run them.
But it's just a thought. I'm okay with both ways.